packages/aws_common/lib/src/logging/aws_logger.dart

```diff
@@ -2,14 +2,20 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import 'dart:async';
+import 'dart:collection';
 
 import 'package:aws_common/aws_common.dart';
 import 'package:aws_common/src/logging/logging_ext.dart';
+import 'package:collection/collection.dart';
 import 'package:logging/logging.dart';
 import 'package:meta/meta.dart';
 
 /// The default log level used by [AWSLogger].
 const zDefaultLogLevel = LogLevel.info;
+String _stateErorrMsgPluginIsRegistered(AWSLogger logger) =>
```

Comment:
Can you make this static on the logger class

Reply:
```suggestion
String _stateErrorMsgPluginIsRegistered(AWSLogger logger) =>
```

Reply:
it can't be static bc need instance data.

Reply:
However the current usage only ever uses `this` for `logger`

Reply:
updated to private property and use Never. 
why do you suggest to use static method and passing logger as parameter?

Reply:
I copied over what you had. If you're only ever reporting on the current logger then you obviously don't need to pass the logger parameter.

Reply:
The point was about keeping it within `AWSLogger` not in the global namespace, and consolidating code to include the `throw` statement.